src/backend/ApiTypes.ts

```diff
@@ -0,0 +1,90 @@
+export interface DirectoryEntry {
+    name?: string,
+    type?: EntryType,
+    mode?: Permissions,
+    size?: number,
+    mtime?: Time,
+    uid?: number,
+    gid?: number,
+    obj?: ObjectID,
+    summ?: DirectorySummary
+}
+
+export interface DirectorySummary {
+    size: number,
+    files: number,
+    symlinks: number,
+    dirs: number,
+    maxTime: Date,
+    incomplete?: string,
+    numFailed: number,
+    numIgnoredErrors?: number,
+    errors?: EntryWithError[]
+}
+
+export interface EntryWithError {
+    path: string,
+    error: string
+}
+
+export type Time = string | number | Date;
+
+export type EntryType = string;
+
+export interface Obj {
+
+}
+
+export interface InfoStruct {
+    contentID: ObjectID,
```

Comment:
all fields are required.

Reply:
As above, took the required info straight from the Go code.